File: back-end/src/main/java/com/aep/catalogo/models/Prestador.java
package com.aep.catalogo.models;

import javax.persistence.*;
import lombok.Data;
import java.util.List;
import java.util.Set;
import java.time.LocalDateTime;
import com.fasterxml.jackson.annotation.JsonGetter;

import com.aep.catalogo.models.enums.CategoriaPrestador;
import com.aep.catalogo.models.enums.SubcategoriaPrestador;

@Data
@Entity
public class Prestador {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    
    @ManyToOne
    @JoinColumn(name = "usuario_id")
    private User usuario;
    
    private String nome;
    private String email;
    private String telefone;
    private String whatsapp;
    
    private String titulo;
    private String descricao;
    
    @Enumerated(EnumType.STRING)
    private CategoriaPrestador categoria;
    
    @ElementCollection
    @Enumerated(EnumType.STRING)
    private Set<SubcategoriaPrestador> subcategorias;
    
    private String cidade;
    private String estado;
    private String endereco;
    private Integer raioAtendimento;
    
    private Double precoHora;
    private Double precoServico;
    private Boolean disponivel = true;
    
    private Double avaliacaoMedia = 0.0;
    private Integer totalAvaliacoes = 0;
    
    private String fotoPerfil;
    
    @JsonGetter("fotoPerfil")
    public String getFotoPerfilCompleta() {
        if (fotoPerfil != null && !fotoPerfil.startsWith("http")) {
            return "http://localhost:3001" + fotoPerfil;
        }
        return fotoPerfil;
    }
    
    @ElementCollection
    private List<String> fotos;
    
    @OneToMany(mappedBy = "prestador", cascade = CascadeType.ALL)
    private List<Servico> servicos;
    
    private LocalDateTime criadoEm = LocalDateTime.now();
    private LocalDateTime atualizadoEm = LocalDateTime.now();
}

File: front-end/src/types/prestador.ts
export enum CategoriaPrestador {
    MANUTENCAO_REPARO = 'manutencao_reparo',
    CASA_LIMPEZA = 'casa_limpeza',
    MODA_COSTURA = 'moda_costura',
    EDUCACAO = 'educacao',
    TECNOLOGIA = 'tecnologia',
    AUTOMOTIVO = 'automotivo',
    BELEZA_BEM_ESTAR = 'beleza_bem_estar'
}

export enum SubcategoriaPrestador {

    MECANICO = 'mecanico',
    ELETRICISTA = 'eletricista',
    ENCANADOR = 'encanador',
    AR_CONDICIONADO = 'ar_condicionado',
    ELETRODOMESTICOS = 'eletrodomesticos',
 
    DIARISTA = 'diarista',
    FAXINEIRA = 'faxineira',
    JARDINEIRO = 'jardineiro',
    PINTOR = 'pintor',


    COSTUREIRA = 'costureira',
    ALFAIATE = 'alfaiate',


    PROFESSOR_PARTICULAR = 'professor_particular',
    REFORCO_ESCOLAR = 'reforco_escolar',


    TECNICO_INFORMATICA = 'tecnico_informatica',


    MECANICO_AUTO = 'mecanico_auto',


    CABELEIREIRO = 'cabeleireiro',
    MANICURE = 'manicure'
}

export interface Prestador {
    id: string;
    usuarioId: string;
    nome: string;
    email: string;
    telefone: string;
    whatsapp?: string

    titulo: string;
    descricao: string;
    categoria: CategoriaPrestador;
    subcategoria: SubcategoriaPrestador[];

    cidade: string;
    estado: string;
    endereco?: string;
    raioAtentimento: number; // Aqui é em Km

    precoHora?: number;
    precoServico?: number;
    disponivel: boolean;

    avaliacaoMedia: number;
    totalAvaliacoes: number;

    fotoPerfil?: string;
    fotos: string[];

    criadoEm: Date;
    atualizadoEm: Date;
}

export interface FiltrosPrestador {
    categoria?: CategoriaPrestador;
    subcategoria?: SubcategoriaPrestador;
    cidade?: string;
    precoMin?: number;
    precoMax?: number;
    avaliacaoMin?: number;
    disponivel?: boolean;
}